src/vs/workbench/api/node/mainThreadEditors.ts

```diff
@@ -102,11 +104,16 @@ export class MainThreadEditors extends MainThreadEditorsShape {
 
 	// --- from extension host process
 
-	$tryShowTextDocument(resource: URI, position: EditorPosition, preserveFocus: boolean): TPromise<string> {
+	$tryShowTextDocument(resource: URI, options: vscode.ShowTextDocumentOptions): TPromise<string> {
+		const position: EditorPosition = TypeConverters.fromViewColumn(options.column);
```

Comment:
@jrieken I moved some of the data massaging into here, so that it could take the same `ShowTextDocumentOptions` (since I think you can technically call the proxy methods from inside an extension).